android/app/src/main/java/org/fossasia/openevent/data/repository/RealmDataRepository.java

```diff
@@ -384,8 +384,8 @@ public Session getSession(String title) {
         return realm.where(Session.class).equalTo("microlocation.name", location).findAllSortedAsync(Session.START_TIME);
     }
 
-    public RealmResults<Session> getSessionsByDate(String date, String sortCriteria) {
-        return realm.where(Session.class).equalTo("startDate", date).findAllSortedAsync(sortCriteria);
```

Comment:
Is there a reason to change this asynchronous sorting?

Reply:
@iamareebjamal The only reason is because this function is only used by DayScheduleFragment and that sorting is already done separately in another function in DayScheduleFragment. Sorting is repeated everytime getSortedSessionsByTracks is called because it is needed for the sorting that we specify using the UI (Ascending, descending, by tracks, by start time etc)  Thus when we retrieve the data from Realm, the sessions do not need to be sorted. 

We can also leave this to return sorted sessions though, if it is potentially used by other fragments/activities. I don't think it'll make a difference for DayScheduleFragment other than efficiency.